src/main/java/hudson/plugins/git/GitStatus.java

```diff
@@ -165,25 +165,31 @@ public void writeBody(PrintWriter w) {
     public static abstract class Listener implements ExtensionPoint {
 
         /**
-         * Called when there is a change notification on a specific repository url.
-         *
-         * @param uri      the repository uri.
-         * @param branches the (optional) branch information.
-         * @return any response contributors for the response to the push request.
-         * @since 1.4.1
-         * @deprecated implement #onNotifyCommit(org.eclipse.jgit.transport.URIish, String, String...)
+         * @deprecated implement {@link #onNotifyCommit(org.eclipse.jgit.transport.URIish, String, List, String...)}
          */
         public List<ResponseContributor> onNotifyCommit(URIish uri, String[] branches) {
-            return onNotifyCommit(uri, null, branches);
+            throw new AbstractMethodError();
```

Comment:
Thus `GitStatus` calls the 4-arg method. If the implementation is new and overrides that, great. If it is a little old and overrides the 3-arg method, the default 4-arg impl will delegate to that, dropping the parameters. If it is even older and overrides the original 2-arg method, the default 3-arg impl will delegate to _that_, dropping also the SHA-1.